### DISCUSSION TITLE
Pass BUFFERS-FILES to org-ql-search links

### DISCUSSION THREAD

**Discussion by fleimgruber** [alphapapa/org-ql#423] [Ideas]
While reading https://github.com/alphapapa/org-ql?tab=readme-ov-file#links I was thinking whether we could pass a list of files (BUFFERS-FILES) to the org-ql-search command in the link?

**Comment by alphapapa** [comment]
When you store a link, it asks about which files to search when the link is opened.  If you want to change the list of files, you can open the link, change the files, and store the link again.

**Comment by fleimgruber** [comment]
Oh I see, I wrote the links manually as mentioned in the 3rd paragraph of that section

**Comment by fleimgruber** [comment]
I hoped for something more readable than:

[[org-ql-search:(tags "the_tag")?buffers-files=%22c%3A%2FUsers%2FLeimgruberF%2Fpath%2Fto%2Ffile.org%22]]

So I guess I will go with this instead:

```org
#+begin_src elisp
(org-ql-search "~/path/to/file.org" "(tags \"the_tag\")")
#+end_src
```

**Reply by alphapapa** [reply]
Links like that are not intended to be human-readable.  They're URL-encoded.  It's intended that you set the title of the search view to communicate which files are searched, and then that title is offered automatically as the description when inserting the link.

It's not intended that such links have no description, because the link itself is not intended to be human-readable.

**Reply by fleimgruber** [reply]
Thanks for explaining!
